packages/server/src/api/controllers/ai.ts

```diff
@@ -0,0 +1,59 @@
+import { env, features } from "@budibase/backend-core"
+import {
+  FeatureFlag,
+  GenerateJsRequest,
+  GenerateJsResponse,
+  UserCtx,
+} from "@budibase/types"
+import OpenAI from "openai"
+
+const MARKDOWN_CODE_BLOCK = /```(?:\w+)?\n([\s\S]+?)\n```/
+
+export async function generateJs(
+  ctx: UserCtx<GenerateJsRequest, GenerateJsResponse>
+) {
+  if (!(await features.isEnabled(FeatureFlag.AI_JS_GENERATION))) {
+    ctx.throw(400, "AI JS generation is disabled")
+  }
+
+  if (!env.OPENAI_API_KEY) {
+    ctx.throw(400, "OpenAI API key is not set")
```

Comment:
These 400 look strange to me. None of them is "bad request", both of them are due server conditions, no because of the request